src/controllers/users.controller.ts

```diff
@@ -60,6 +60,18 @@ class UsersController {
       next(error);
     }
   };
+
+  public manageGroups = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const data = req.body;
+      const userId = Number(req.params.id);
+      const groupData = await this.userService.manageGroups(userId, data);
```

Comment:
I suggest some validation here. Refer to [this comment](https://github.com/NdagiStanley/soul-refiner/pull/1/files#r2133677005).